### PR TITLE
chore(ION Identity client): rename to IONIdentity

### PR DESCRIPTION
**Pull request by ice-tychon**

### CODE REVIEW

File: packages/ion_identity_client/lib/src/users/ion_connect_indexers/data_sources/ion_connect_indexers_data_source.dart [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      throw const UnauthenticatedException();
+    }
+
+    final response = await _networkClient.get(
+      '$getUserConnectIndexersPath/$userId/ion-connect-indexers',
+      headers: RequestHeaders.getAuthorizationHeaders(
+        username: username,
+        token: token.token,
+      ),
+      decoder: IonConnectIndexersResponse.fromJson,
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/lib/src/users/ion_connect_indexers/data_sources/ion_connect_indexers_data_source.dart]: Fixed

File: packages/ion_identity_client/lib/src/users/ion_connect_indexers/data_sources/ion_connect_indexers_data_source.dart [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    this._networkClient,
+    this._tokenStorage,
+  );
+
+  final NetworkClient _networkClient;
+  final TokenStorage _tokenStorage;
+
+  static const getUserConnectIndexersPath = '/v1/users';
+
+  Future<List<String>> fetchIonConnectIndexers({
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/lib/src/users/ion_connect_indexers/data_sources/ion_connect_indexers_data_source.dart]: Fixed.

File: packages/ion_identity_client/lib/src/users/ion_connect_indexers/data_sources/ion_connect_indexers_data_source.dart [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+class IonConnectIndexersDataSource {
+  IonConnectIndexersDataSource(
+    this._networkClient,
+    this._tokenStorage,
+  );
+
+  final NetworkClient _networkClient;
+  final TokenStorage _tokenStorage;
+
+  static const getUserConnectIndexersPath = '/v1/users';
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/lib/src/users/ion_connect_indexers/data_sources/ion_connect_indexers_data_source.dart:17]: Fixed.

File: packages/ion_identity_client/lib/src/users/ion_connect_indexers/data_sources/ion_connect_indexers_data_source.dart [hunk 4]
```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:ion_identity_client/ion_identity.dart';
+import 'package:ion_identity_client/src/core/network/network_client.dart';
+import 'package:ion_identity_client/src/core/token_storage/token_storage.dart';
+import 'package:ion_identity_client/src/core/types/request_headers.dart';
+
+class IonConnectIndexersDataSource {
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/lib/src/users/ion_connect_indexers/data_sources/ion_connect_indexers_data_source.dart]: Fixed.

File: packages/ion_identity_client/lib/src/core/service_locator/ion_identity_clients/wallets_client_service_locator.dart [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+class WalletsClientServiceLocator {
+  factory WalletsClientServiceLocator() {
+    return _instance;
+  }
+
+  WalletsClientServiceLocator._internal();
+
+  static final WalletsClientServiceLocator _instance = WalletsClientServiceLocator._internal();
+
+  IONWallets wallets({
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/lib/src/core/service_locator/ion_identity_clients/wallets_client_service_locator.dart:32]: Fixed.

File: packages/ion_identity_client/lib/src/auth/ion_auth.dart [hunk 6]
```diff
@@ -10,15 +10,15 @@ import 'package:ion_identity_client/src/core/token_storage/token_storage.dart';
 
 /// A class that handles user authentication processes, including user registration,
 /// login, and logout.
-class IonAuth {
-  /// Constructs an [IonAuth] instance with the provided services and token storage.
+class IONAuth {
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/lib/src/auth/ion_auth.dart:13]: Fixed.

File: packages/ion_identity_client/example/lib/pages/wallet_assets/wallet_assets_page.dart [hunk 7]
```diff
@@ -67,7 +67,7 @@ class _Body extends HookConsumerWidget {
 
   String _getErrorMessage(BuildContext context, Object error) {
     return switch (error) {
-          IonException(:final message) => message,
+          IONException(:final message) => message,
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/example/lib/pages/wallet_assets/wallet_assets_page.dart:70]: Fixed.

File: packages/ion_identity_client/example/lib/pages/set_ion_connect_relays/providers/set_ion_connect_relays_notifier.dart [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+part 'set_ion_connect_relays_notifier.g.dart';
+
+@riverpod
+class SetIonConnectRelaysNotifier extends _$SetIonConnectRelaysNotifier {
+  @override
+  FutureOr<SetIonConnectRelaysResponse?> build() async {
+    return null;
+  }
+
+  Future<void> setIonConnectRelays({
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/example/lib/pages/set_ion_connect_relays/providers/set_ion_connect_relays_notifier.dart]: Fixed.

File: packages/ion_identity_client/example/lib/pages/get_ion_connect_indexers/providers/ion_connect_indexers_notifier.dart [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+part 'ion_connect_indexers_notifier.g.dart';
+
+@riverpod
+class IonConnectIndexersNotifier extends _$IonConnectIndexersNotifier {
+  @override
+  FutureOr<List<String>?> build() async {
+    return null;
+  }
+
+  Future<void> fetchIonConnectIndexers(String userId) async {
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/example/lib/pages/get_ion_connect_indexers/providers/ion_connect_indexers_notifier.dart]: Fixed.

File: packages/ion_identity_client/lib/src/core/service_locator/clients_service_locator.dart [hunk 10]
```diff
@@ -1,399 +1,48 @@
...
 
   static final ClientsServiceLocator _instance = ClientsServiceLocator._internal();
-}
 
-mixin _IonClient {
-  final Map<String, IonApiUserClient> _clients = {};
+  final Map<String, IONIdentityUserClient> _clients = {};
 
-  IonApiUserClient getIonClient({
+  IONIdentityUserClient ionIdentityUserClient({
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/lib/src/core/service_locator/clients_service_locator.dart]: Fixed.

**Review by ice-alcides — APPROVED** [review]

**Review by ice-kreios — APPROVED** [review]

File: packages/ion_identity_client/lib/src/core/service_locator/ion_identity_clients/auth_client_service_locator.dart [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+class AuthClientServiceLocator {
+  factory AuthClientServiceLocator() {
+    return _instance;
+  }
+
+  AuthClientServiceLocator._internal();
+
+  static final AuthClientServiceLocator _instance = AuthClientServiceLocator._internal();
+
+  IONAuth auth({
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/lib/src/core/service_locator/ion_identity_clients/auth_client_service_locator.dart]: Fixed.

File: packages/ion_identity_client/lib/src/core/service_locator/ion_identity_clients/users_client_service_locator.dart [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+  }) =>
+      UserDetailsService(
+        username,
+        UserDetailsDataSource(
+          IONIdentityServiceLocator.networkClient(config: config),
+          IONIdentityServiceLocator.tokenStorage(),
+        ),
+      );
+
+  IonConnectIndexersService _ionConnectIndexers({
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/lib/src/core/service_locator/ion_identity_clients/users_client_service_locator.dart]: Fixed.

File: packages/ion_identity_client/lib/src/core/service_locator/ion_identity_clients/users_client_service_locator.dart [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+  }) =>
+      IonConnectIndexersService(
+        username,
+        IonConnectIndexersDataSource(
+          IONIdentityServiceLocator.networkClient(config: config),
+          IONIdentityServiceLocator.tokenStorage(),
+        ),
+      );
+
+  SetIonConnectRelaysService _setIonConnectRelays({
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/lib/src/core/service_locator/ion_identity_clients/users_client_service_locator.dart]: Fixed.

File: packages/ion_identity_client/lib/src/core/service_locator/ion_identity_clients/users_client_service_locator.dart [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+class UsersClientServiceLocator {
+  factory UsersClientServiceLocator() {
+    return _instance;
+  }
+
+  UsersClientServiceLocator._internal();
+
+  static final UsersClientServiceLocator _instance = UsersClientServiceLocator._internal();
+  
+  IONUsers users({
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/lib/src/core/service_locator/ion_identity_clients/users_client_service_locator.dart]: Fixed.

**Pull request merged**